pyuvdata/uvcal/tests/test_calfits.py

```diff
@@ -278,8 +312,55 @@ def test_extra_keywords(tmp_path):
 
     assert cal_in == cal_out
 
+    return
+
 
-def test_input_flag_array(tmp_path):
+def test_extra_keywords_errors(tmp_path):
+    cal_in = UVCal()
+    calfits_file = os.path.join(DATA_PATH, "zen.2457698.40355.xx.gain.calfits")
+    testfile = str(tmp_path / "outtest_omnical.fits")
+    cal_in.read_calfits(calfits_file)
+
+    # check for warnings & errors with extra_keywords that are dicts, lists or arrays
+    cal_in.extra_keywords["testdict"] = {"testkey": 23}
+    uvtest.checkWarnings(
+        cal_in.check, message=["testdict in extra_keywords is a list, array or dict"]
+    )
+    pytest.raises(TypeError, cal_in.write_calfits, testfile, run_check=False)
+    cal_in.extra_keywords.pop("testdict")
+
+    cal_in.extra_keywords["testlist"] = [12, 14, 90]
+    uvtest.checkWarnings(
+        cal_in.check, message=["testlist in extra_keywords is a list, array or dict"]
+    )
+    pytest.raises(TypeError, cal_in.write_calfits, testfile, run_check=False)
+    cal_in.extra_keywords.pop("testlist")
+
+    cal_in.extra_keywords["testarr"] = np.array([12, 14, 90])
+    uvtest.checkWarnings(
+        cal_in.check, message=["testarr in extra_keywords is a list, array or dict"]
+    )
+    pytest.raises(TypeError, cal_in.write_calfits, testfile, run_check=False)
+    cal_in.extra_keywords.pop("testarr")
+
+    # check for warnings with extra_keywords keys that are too long
+    cal_in.extra_keywords["test_long_key"] = True
+    uvtest.checkWarnings(
+        cal_in.check,
+        message=["key test_long_key in extra_keywords is longer than 8 characters"],
+    )
+    uvtest.checkWarnings(
+        cal_in.write_calfits,
+        [testfile],
+        {"run_check": False, "clobber": True},
+        message=["key test_long_key in extra_keywords is longer than 8 characters"],
+    )
+    cal_in.extra_keywords.pop("test_long_key")
+
+    return
```

Comment:
With all the dict-key `pop`ing going on here and adding extra keywords, this might be another reasonable case for a parametrize call.

Reply:
Good point, I can atomize/parametrize this test.

Reply:
This is done now.